### PR TITLE
WITSML Change Mechanism new feature documentation

### PR DESCRIPTION
**Pull request by devfabiosilva**

### CODE REVIEW

File: collector/configuration/liverig.properties.md [hunk 1]
```diff
@@ -68,7 +68,14 @@
...
-| `enable_revision`                                                                                                                                                                                                                                                                                                                                  | `boolean`  | `false`                              | 2.24.1 |           |                                                                                                          |                          |
+| `witsml_change_mechanism_feature` <br> Enable/Disable WITSML change mechanism. | `boolean` | `true` | 5.17.x |||||
```
Comment (amadeubarbosa) [inline collector/configuration/liverig.properties.md]: The default value is not true as far I know. Please fix this.

File: collector/configuration/liverig.properties.md [hunk 2]
```diff
@@ -68,7 +68,14 @@
...
+| `witsml_change_mechanism_delay_period_in_seconds` <br> Delay period of WITSML change mechanism. | `int` | `0` | 5.17.x |s||||
+| `witsml_change_mechanism_detection_period_in_seconds` <br> WITSML change detection period mechanism. | `int` | `10` | 5.17.x |s||||
+| `witsml_change_mechanism_initial_last_change_feature` <br> Enable/Disable WITSML initial last change mechanism. | `boolean` | `true` | 5.17.x |||||
+| `witsml_change_mechanism_late_channels_feature` <br> Enable/Disable WITSML late channel change mechanism. | `boolean` | `true` | 5.17.x |||||
+| `witsml_change_mechanism_object_growing_feature` <br> Enable/Disable WITSML object growing change mechanism. | `boolean` | `true` | 5.17.x |||||
+| `witsml_change_mechanism_initial_last_change_in_minutes` <br> WITSML last change in minutes. | `int` | `1` | 5.17.x | minutes ||||
```
Comment (amadeubarbosa) [inline collector/configuration/liverig.properties.md]: The default values are present in AppProperties.java like the following:
```java
    public static final boolean DEFAULT_WITSML_CHANGE_MECHANISM_FEATURE = false;
    public static final boolean DEFAULT_WITSML_CHANGE_MECHANISM_LATE_CHANNELS_FEATURE = false;
    public static final boolean DEFAULT_WITSML_CHANGE_MECHANISM_OBJECT_GROWING_FEATURE = false;
    public static final long DEFAULT_WITSML_CHANGE_MECHANISM_DETECTION_PERIOD_IN_SECONDS = 10;
    public static final boolean DEFAULT_WITSML_CHANGE_MECHANISM_INITIAL_LAST_CHANGE_FEATURE = true;
    public static final long DEFAULT_WITSML_CHANGE_MECHANISM_INITIAL_LAST_CHANGE_IN_MINUTES = 1440;
    public static final long DEFAULT_WITSML_CHANGE_DELAY_PERIOD_IN_SECONDS = 0;
```

please fix the documentation here to reflect the correct values.

File: collector/configuration/liverig.properties.md [hunk 3]
```diff
@@ -134,4 +141,13 @@ wildcard_period_in_seconds=600
...
 
+# WITSML Change mechanism features
+#
+# witsml_change_mechanism_feature=true
+# witsml_change_mechanism_feature.witsml_change_mechanism_delay_period_in_seconds=2
+# witsml_change_mechanism_feature.witsml_change_mechanism_detection_period_in_seconds=10
+# witsml_change_mechanism_feature.witsml_change_mechanism_initial_last_change_feature=false
+# witsml_change_mechanism_feature.witsml_change_mechanism_late_channels_feature=true
+# witsml_change_mechanism_feature.witsml_change_mechanism_object_growing_feature=true
+# witsml_change_mechanism_feature.witsml_change_mechanism_initial_last_change_in_minutes=10
```
Comment (amadeubarbosa) [inline collector/configuration/liverig.properties.md]: Please fix here also.. probably you copied from "test plan" routine but not from the original source code.

Here it's not a documentation how to enable the new feature but the documentation about the default values and property names

```java
    public static final boolean DEFAULT_WITSML_CHANGE_MECHANISM_FEATURE = false;
    public static final boolean DEFAULT_WITSML_CHANGE_MECHANISM_LATE_CHANNELS_FEATURE = false;
    public static final boolean DEFAULT_WITSML_CHANGE_MECHANISM_OBJECT_GROWING_FEATURE = false;
    public static final long DEFAULT_WITSML_CHANGE_MECHANISM_DETECTION_PERIOD_IN_SECONDS = 10;
    public static final boolean DEFAULT_WITSML_CHANGE_MECHANISM_INITIAL_LAST_CHANGE_FEATURE = true;
    public static final long DEFAULT_WITSML_CHANGE_MECHANISM_INITIAL_LAST_CHANGE_IN_MINUTES = 1440;
    public static final long DEFAULT_WITSML_CHANGE_DELAY_PERIOD_IN_SECONDS = 0;
```

File: collector/configuration/liverig.properties.md [hunk 4]
```diff
@@ -68,13 +68,13 @@
...
+| `witsml_change_mechanism_feature` <br> Enables a mechanism for identifying change for data objects in a WITSML server | `boolean` | `false` | 5.17.x |||||
+| `witsml_change_mechanism_delay_period_in_seconds` <br> The number of seconds for the client to detect a change in a growing data object | `int` | `0` | 5.17.x |s||||
+| `witsml_change_mechanism_detection_period_in_seconds` <br> The number of seconds for the client to detect a change in a growing data object | `int` | `10` | 5.17.x |s||||
+| `witsml_change_mechanism_initial_last_change_feature` <br> Enables the use of a default initial lastChange when starts the feature | `boolean` | `true` | 5.17.x |||||
+| `witsml_change_mechanism_late_channels_feature` <br> Allows to collect channels that arrive with low-frequency samples | `boolean` | `false` | 5.17.x |||||
+| `witsml_change_mechanism_object_growing_feature` <br> Enables the use of object-growing queries when there aren't changed objects | `boolean` | `false` | 5.17.x |||||
+| `witsml_change_mechanism_initial_last_change_in_minutes` <br> he number of minutes before now for use as the default last change for an initial request | `int` | `1440` | 5.17.x | minutes ||||
```
Comment (jonyfs) [inline collector/configuration/liverig.properties.md]: Please, fix the text adding the letter **T**

File: collector/configuration/liverig.properties.md (hunk 1 above)
Comment (devfabiosilva) [reply collector/configuration/liverig.properties.md]: Changed to _false_

File: collector/configuration/liverig.properties.md [hunk 5]
```diff
@@ -68,7 +68,14 @@
...
-| `enable_revision`                                                                                                                                                                                                                                                                                                                                  | `boolean`  | `false`                              | 2.24.1 |           |                                                                                                          |                          |
+| `witsml_change_mechanism_feature` <br> Enables a mechanism for identifying change for data objects in a WITSML server | `boolean` | `false` | 5.17.x |||||
```
Comment (efsh) [inline collector/configuration/liverig.properties.md:71]: It is important to mention that this new feature disables the realTime sync feature

File: collector/configuration/liverig.properties.md [hunk 6]
```diff
@@ -68,7 +68,14 @@
...
+| `witsml_change_mechanism_delay_period_in_seconds` <br> The number of seconds for the client to detect a change in a growing data object | `int` | `0` | 5.17.x |s||||
+| `witsml_change_mechanism_detection_period_in_seconds` <br> The number of seconds for the client to detect a change in a growing data object | `int` | `10` | 5.17.x |s||||
```
Comment (efsh) [inline collector/configuration/liverig.properties.md]: Is the same description for both properties?

File: collector/configuration/liverig.properties.md [hunk 7]
```diff
@@ -68,7 +68,14 @@
...
+| `witsml_change_mechanism_initial_last_change_feature` <br> Enables the use of a default initial lastChange when starts the feature | `boolean` | `true` | 5.17.x |||||
+| `witsml_change_mechanism_late_channels_feature` <br> Allows to collect channels that arrive with low-frequency samples | `boolean` | `false` | 5.17.x |||||
```
Comment (efsh) [inline collector/configuration/liverig.properties.md]: keep in mind that this feature only works if the `mechanism_feature` is enabled

**Comment by efsh** [comment]
A better PR's title is welcomed

File: collector/configuration/liverig.properties.md (hunk 5 above)
Comment (devfabiosilva) [reply collector/configuration/liverig.properties.md:71]: @efsh It is already mentioned in [new issue](https://github.com/intelie/gitbook-drilling/pull/55/commits/ffe55291d8b44b1b51c4ece3aeb829cd4b95a0bd#diff-10fe8dbb5f5ccc642a293b18881dd2ac37782b82f62411dacf6b165f8b60e3caR11) for #55

**Comment by efsh** [comment]
LGTM @amadeubarbosa @jonyfs

**Pull request merged**